server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java

```diff
@@ -367,4 +370,63 @@ private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint,
     private NRTReplicationEngine buildNrtReplicaEngine(AtomicLong globalCheckpoint, Store store) throws IOException {
         return buildNrtReplicaEngine(globalCheckpoint, store, defaultSettings);
     }
+
+    public void testGetSegmentInfosSnapshotPreservesFilesUntilRelease() throws Exception {
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
+
+        // TODO: Remove this divergent commit logic and copy Segments_N from primary with node-node.
```

Comment:
Stale comment based on https://github.com/opensearch-project/OpenSearch/issues/9016 where we are discussing to not copy segment_N from primary/remote store.

Reply:
This is not stale *yet* because we stil have the divergent logic in the engine.  Will be removing this in the next pr.